src/AccessibilityInsights.SharedUxTests/Settings/InstallationInfoTests.cs

```diff
@@ -3,57 +3,152 @@
+                    Assert.AreEqual(date, info.LastReset);
+                    Assert.AreEqual(guid, info.InstallationGuid);
+                }
             }
         }
-#endif
+
+        [TestMethod]
+        [Timeout(1000)]
+        public void LoadFromPath_FileDoesNotExist_LastResetIsProvidedTime()
+        {
+            DateTime now = January2015;
+            bool wasRefreshed = false;
+            InstallationInfo.WriteToDiskOverride = (_, __) => wasRefreshed = true;
+
+            InstallationInfo info = InstallationInfo.LoadFromPath(_testFolder, now);
+
+            Assert.IsTrue(wasRefreshed);
+            Assert.AreEqual(now, info.LastReset);
+            Assert.AreNotEqual(InstallationGuid, info.InstallationGuid);
+        }
+
+        [TestMethod]
+        [Timeout(1000)]
+        public void LoadFromPath_FileIsCurrent_FileIsNotRefreshed()
+        {
+            DateTime now = January2015;
+            bool wasRefreshed = false;
+            InstallationInfo.WriteToDiskOverride = (_, __) => wasRefreshed = true;
+            InstallationInfo.ReadFromDiskOverride = (_) => new InstallationInfo(InstallationGuid, now);
+            InstallationInfo info = InstallationInfo.LoadFromPath(_testFolder, now);
+
+            Assert.IsFalse(wasRefreshed);
+            Assert.AreEqual(now, info.LastReset);
+            Assert.AreEqual(InstallationGuid, info.InstallationGuid);
+        }
+
+        [TestMethod]
+        [Timeout(1000)]
+        public void LoadFromPath_FileIsOneMonthOld_FileIsRefreshedWithProvidedTime()
+        {
+            DateTime now = February2015;
+            bool wasRefreshed = false;
+            InstallationInfo.WriteToDiskOverride = (_, __) => wasRefreshed = true;
+            InstallationInfo.ReadFromDiskOverride = (_) => new InstallationInfo(InstallationGuid, January2015);
+
+            InstallationInfo info = InstallationInfo.LoadFromPath(_testFolder, now);
+
+            Assert.IsTrue(wasRefreshed);
+            Assert.AreEqual(now, info.LastReset);
+            Assert.AreNotEqual(InstallationGuid, info.InstallationGuid);
+        }
+
+        [TestMethod]
+        [Timeout(1000)]
+        public void LoadFromPath_FileIsOneYearOld_FileIsRefreshedWithProvidedTime()
+        {
+            DateTime now = January2016;
+            bool wasRefreshed = false;
+            InstallationInfo.WriteToDiskOverride = (_, __) => wasRefreshed = true;
+            InstallationInfo.ReadFromDiskOverride = (_) => new InstallationInfo(InstallationGuid, January2015);
+
+            InstallationInfo info = InstallationInfo.LoadFromPath(_testFolder, now);
+
+            Assert.IsTrue(wasRefreshed);
+            Assert.AreEqual(now, info.LastReset);
+            Assert.AreNotEqual(InstallationGuid, info.InstallationGuid);
+        }
+
+        [TestMethod]
+        [Timeout(1000)]
+        public void LoadFromPath_FileIsOneMonthOld_UnableToWrite_ReturnsRefreshedDataEachTime()
+        {
+            DateTime now = February2015;
```

Comment:
```suggestion
            // Intentionally don't set WriteToDiskOverride so that we try to write to disk. Since we
            // haven't created a folder for _testFolder, the attempt to write to disk will fail.
            DateTime now = February2015;
```